internal/artifacts/module_arch_test.go

```diff
@@ -89,97 +90,142 @@ builders:
 			})
-			It("required module m2 has ready artifact 'test2.txt' and creates a new one 'test2_copy.txt'", func() {
-				Ω(ExecuteSoloBuild(getTestPath("mtaModelsBuild"), getResultPath(), nil, []string{"m1", "m3"}, true, os.Getwd)).Should(Succeed())
+
+			It("required module m2 has ready artifact 'test2.txt' and creates a new one 'test2_copy.txt', mtad.yaml generated", func() {
+				Ω(ExecuteSoloBuild(getTestPath("mtaModelsBuild"), getResultPath(), nil, []string{"m1", "m3"}, true, true, "cf", os.Getwd, yaml.Marshal)).Should(Succeed())
 				Ω(getTestPath("result", "data.zip")).Should(BeAnExistingFile())
 				Ω(getTestPath("result", "m3.zip")).Should(BeAnExistingFile())
 				validateArchiveContents([]string{"test.txt", "test2.txt", "test2_copy.txt"}, getTestPath("result", "data.zip"))
+				mtadContent, err := ioutil.ReadFile(getTestPath("result", "mtad.yaml"))
+				Ω(err).Should(Succeed())
+				mtadObj, err := mta.Unmarshal(mtadContent)
+				Ω(err).Should(Succeed())
+				for _, module := range mtadObj.Modules {
+					if module.Name == "m1" {
+						Ω(module.Path).Should(Equal(getTestPath("result", "data.zip")))
+					} else if module.Name == "m3" {
+						Ω(module.Path).Should(Equal(getTestPath("result", "m3.zip")))
+					}
+				}
+			})
+
+			Describe("mtad.yaml generation fails on corrupted platform configuration", func() {
+
+				var platformConfig []byte
+
+				BeforeEach(func() {
+					platformConfig = platform.PlatformConfig
+					platform.PlatformConfig = []byte("corrupted")
+				})
+
+				AfterEach(func() {
+					platform.PlatformConfig = platformConfig
+				})
+
+				It("failure", func() {
+					Ω(ExecuteSoloBuild(getTestPath("mtaModelsBuild"), getResultPath(), nil, []string{"m1", "m3"}, true, true, "cf", os.Getwd, yaml.Marshal)).Should(HaveOccurred())
+				})
+			})
+
+			It("required module m2 has ready artifact 'test2.txt' and creates a new one 'test2_copy.txt', mtad.yaml not generated", func() {
+				Ω(ExecuteSoloBuild(getTestPath("mtaModelsBuild"), getResultPath(), nil, []string{"m1"}, true, false, "cf", os.Getwd, yaml.Marshal)).Should(Succeed())
+				Ω(getTestPath("result", "data.zip")).Should(BeAnExistingFile())
+				Ω(getTestPath("result", "m3.zip")).ShouldNot(BeAnExistingFile())
+				Ω(getTestPath("result", "mtad.yaml")).ShouldNot(BeAnExistingFile())
+				validateArchiveContents([]string{"test.txt", "test2.txt", "test2_copy.txt"}, getTestPath("result", "data.zip"))
 			})
+
 			It("modules m1 and m2 have conflicting build results", func() {
-				err := ExecuteSoloBuild(getTestPath("mtaModelsBuild"), getResultPath(), nil, []string{"m1", "m2"}, true, os.Getwd)
+				err := ExecuteSoloBuild(getTestPath("mtaModelsBuild"), getResultPath(), nil, []string{"m1", "m2"}, true, false, "", os.Getwd, yaml.Marshal)
 				Ω(err).Should(HaveOccurred())
 				Ω(err.Error()).Should(ContainSubstring(fmt.Sprintf(multiBuildWithPathsConflictMsg, "m2", "m1", getResultPath(), "data.zip")))
 			})
+
+			It("fails on invalid platform", func() {
```

Comment:
Add `when mtad.yaml should be generated` to the test description

Reply:
added